### ISSUE TITLE
Filtering parameters does not work with Phoenix's special syntax {:keep, attrs}

### ISSUE THREAD

**Issue by philss** [appsignal/appsignal-elixir#498]
## Describe the bug

We are trying to configure parameter filtering for our controllers in Phoenix and we got an error while using the [special syntax](https://hexdocs.pm/phoenix/Phoenix.Logger.html#module-parameter-filtering) `{:keep, attributes}` that Phoenix supports.

For what I could understand, the problem is related to Appsignal considering that this configuration can only be a list, as we can see here: https://github.com/appsignal/appsignal-elixir/blob/cb71636018caa88e7b66e909fcb2b1f0f3dc9359/lib/appsignal/utils/map_filter.ex#L8-L9

Here is the Phoenix implementation for reference: https://github.com/phoenixframework/phoenix/blob/4d61eacd83d61899a895167715d8754c5c8187be/lib/phoenix/logger.ex#L63

## To Reproduce

- Using AppSignal for Elixir package version "1.10.8"
- In my app using Phoenix framework version "1.4.8"
- In this code:
  ```elixir
  # in config/config.exs or prod/release file.
  config :phoenix, :filter_parameters, {:keep, ["id", "meta", "type"]}
  ```

**Comment by philss** [comment]
This is the error we got (redacted):

```
Erlang error: {{%ArgumentError{message: "argument error"}, [{:erlang, :++, [{:keep, ["id", "meta", "type"]}, []], []}, {Appsignal.Utils.MapFilter, :get_filter_parameters, 0, [file: 'lib/appsignal/utils/map_filter.ex', line: 8]}, {Appsignal.Plug, :extract_sample_data, 1, [file: 'lib/appsignal/plug.ex', line: 122]}, {Appsignal.Transaction, :set_request_metadata, 2, [file: 'lib/appsignal/transaction.ex', line: 494]}, {Appsignal.Plug, :finish_with_conn, 2, [file: 'lib/appsignal/plug.ex', line: 68]}, {BrokerWeb.Endpoint, :call, 2, [file: 'lib/broker_web/endpoint.ex', line: 1]}, {Phoenix.Endpoint.Cowboy2Handler, :init, 2, [file: 'lib/phoenix/endpoint/cowboy2_handler.ex', line: 33]}, {:cowboy_handler, :execute, 2, [file: '/opt/app/deps/cowboy/src/cowboy_handler.erl', line: 41]}]}, {BrokerWeb.Endpoint, :call, [%Plug.Conn{adapter: {Plug.Cowboy.Conn, :...}, assigns: %{}, before_send: [], body_params: %Plug.Conn.Unfetched{aspect: :body_params}, cookies: %Plug.Conn.Unfetched{aspect: :cookies}, halted: false, host: "our-host-here", method: "POST", owner: #PID<0.4894.0>, params: %Plug.Conn.Unfetched{aspect: :params}, path_info: ["clients"], path_params: %{}, port: 80, private: %{}, query_params: %Plug.Conn.Unfetched{aspect: :query_params}, query_string: "", remote_ip: {x, x, x, x}, req_cookies: %Plug.Conn.Unfetched{aspect: :cookies}, req_headers: [{"accept", "application/vnd.api+json"}, {"content-length", "1362"}, {"content-type", "application/json"}, {"host", "our-host-here"}, {"user-agent", "curl/7.54.0"}, {"x-forwarded-for", "x.x.x.x"}, {"x-forwarded-host", "our-host-here"}, {"x-forwarded-port", "443"}, {"x-forwarded-proto", "https"}, {"x-original-uri", "/clients"}, {"x-real-ip", "original-ip-here"}, {"x-request-id", "9ddb3901d299f4227483dba20affd87b"}, {"x-scheme", "https"}], request_path: "/clients", resp_body: nil, resp_cookies: %{}, resp_headers: [{"cache-control", "max-age=0, private, must-revalidate"}], scheme: :http, script_name: [], secret_key_base: nil, state: :unset, status: nil}, []]}}
```

And this is the stacktrace (which does not tell much):

```
(phoenix) lib/phoenix/endpoint/cowboy2_handler.ex:42: Phoenix.Endpoint.Cowboy2Handler.init/2
(cowboy) /opt/app/deps/cowboy/src/cowboy_handler.erl:41: :cowboy_handler.execute/2
(cowboy) /opt/app/deps/cowboy/src/cowboy_stream_h.erl:296: :cowboy_stream_h.execute/3
(cowboy) /opt/app/deps/cowboy/src/cowboy_stream_h.erl:274: :cowboy_stream_h.request_process/3
(stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
```

**Comment by philss** [comment]
I'm working on a fix for this issue.

**Comment by jeffkreeftmeijer** [comment]
For anyone reading along; We're reviewing @philss' pull request which includes a patch for this issue in #499. We’re working on getting a patch release out for this. If you run into this in the meantime, could you try Philip’s branch, and let us know how it worked?

    {:appsignal, github: "philss/appsignal-elixir", branch: "support-phoenix-params-filtering"}

**Comment by tombruijn** [comment]
Fixed in #499. Closing.
